crates/wiggle/borrow/src/lib.rs

```diff
@@ -1,11 +1,12 @@
-use std::cell::RefCell;
 use std::collections::HashMap;
+use std::sync::Mutex;
 use wiggle::{BorrowHandle, GuestError, Region};
 
 pub struct BorrowChecker {
     /// Unfortunately, since the terminology of std::cell and the problem domain of borrow checking
     /// overlap, the method calls on this member will be confusing.
-    bc: RefCell<InnerBorrowChecker>,
+    // TODO
+    bc: Mutex<InnerBorrowChecker>,
```

Comment:
FWIW I think this is a real open question. I don't know what to do about this really. The way that `GuestPtr` works and how we need it `Send` and `Sync` necessitates this `Mutex` for now, but there's no threads here so there's no reason this *should* have a mutex. I'm hoping we can basically leave this to get improved later, probably by rethinking the way `GuestPtr` works.

Reply:
(I think that `GuestPtr` needs to be rethought for other reasons too)

Reply:
I'm interested to hear more of your thoughts on this when you have some time!

Reply:
Ok I talked with @pchickey about this and our conclusion is that eventually `GuestPtr` will probably get phased out entirely. Instead of `GuestPtr` we'll generate the wiggle traits with raw borrowed strings already. This means that the `BorrowChecker` here can live in a mutable location and won't need interior mutability, alleviating the need for this `Mutex`. That'll happen in a future PR though